sn_client/src/wallet/mod.rs

```diff
@@ -6,30 +6,56 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-pub(crate) mod send_client;
-pub(crate) mod verifying_client;
-
 use super::Client;
 
+use sn_transfers::{
+    client_transfers::TransactionOutputs,
+    wallet::{Error, LocalWallet, Result},
+};
+
+use futures::future::join_all;
 use sn_dbc::{Dbc, PublicAddress, Token};
-use sn_transfers::wallet::{Error, LocalWallet, Result, SendWallet};
 
 /// A wallet client can be used to send and
 /// receive tokens to/from other wallets.
-pub struct WalletClient<W: SendWallet> {
+pub struct WalletClient {
     client: Client,
-    wallet: W,
+    wallet: LocalWallet,
+    /// These have not yet been successfully confirmed in
+    /// the network and need to be republished, to reach network validity.
+    /// We maintain the order they were added in, as to republish
+    /// them in the correct order, in case any later spend was
+    /// dependent on an earlier spend.
+    unconfirmed_txs: Vec<TransactionOutputs>,
 }
 
-impl<W: SendWallet> WalletClient<W> {
+impl WalletClient {
     /// Create a new wallet client.
-    pub fn new(client: Client, wallet: W) -> Self {
-        Self { client, wallet }
+    pub fn new(client: Client, wallet: LocalWallet) -> Self {
+        Self {
+            client,
+            wallet,
+            unconfirmed_txs: vec![],
+        }
     }
 
     /// Send tokens to another wallet.
     pub async fn send(&mut self, amount: Token, to: PublicAddress) -> Result<Dbc> {
-        let dbcs = self.wallet.send(vec![(amount, to)], &self.client).await?;
+        // retry previous failures
+        self.resend_pending_txs().await;
+
+        // offline transfer
+        let transfer = self.wallet.local_send(vec![(amount, to)]).await?;
+        let dbcs = transfer.created_dbcs.clone();
+
+        // send to network
+        println!("Sending transfer to the network: {transfer:#?}");
+        if let Err(error) = self.client.send(transfer.clone()).await {
+            println!("The transfer was not successfully registered in the network: {error:?}. It will be retried later.");
+            self.unconfirmed_txs.push(transfer);
+        }
+
+        // return created DBCs even if network part failed???
         match &dbcs[..] {
             [info, ..] => Ok(info.dbc.clone()),
             [] => Err(Error::CouldNotSendTokens(
```

Comment:
Left that behaviour as is but must admit I'm not too sure about this so left a comment, anyone know why we do this? (The wallet is already mutated at this point but the network might not know anything as of yet)